src/Codec/CodecLibrary.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+namespace MongoDB\Codec;
+
+use MongoDB\Exception\UnexpectedValueException;
+
+use function array_map;
+use function get_debug_type;
+use function sprintf;
+
+class CodecLibrary implements Codec
+{
+    /** @var array<Decoder> */
+    private $decoders = [];
+
+    /** @var array<Encoder> */
+    private $encoders = [];
+
+    /** @param Decoder|Encoder $items */
+    public function __construct(...$items)
+    {
+        array_map(
+            function ($item) {
+                if ($item instanceof Decoder) {
+                    $this->attachDecoder($item);
+                }
+
+                if ($item instanceof Encoder) {
+                    $this->attachEncoder($item);
+                }
+
+                // Yes, we'll silently discard everything. Please let me already have union types...
```

Comment:
We may want to replace this with an `InvalidArgumentException` - for now this is fine.